installer/install.sh

```diff
@@ -0,0 +1,246 @@
+                ask_for_value "Path to the certificate key" HTTPS_CERT_KEY
+            fi
+        fi
+
+        if [ ! -x /usr/sbin/nginx ]; then
+            ask_yes_no "You don't have nginx installed. Install it" INSTALL_NGINX
+        fi
+
+        while
+            ask_for_value "nginx IPv4 listen address (\"none\" to disable)" LISTEN_IPV4
+            ask_for_value "nginx IPv6 listen address (\"none\" to disable)" LISTEN_IPV6
+            [ "$LISTEN_IPV4" == none ] && [ "$LISTEN_IPV6" == none ]
+        do
+            echo "You need to listen to at least one IP address!"
+
+            # Reset them
+            LISTEN_IPV4="$(get_nginx_ip 4)"
+            LISTEN_IPV6="$(get_nginx_ip 6)"
+        done
+
+    fi
+
+    ask_sudo_pw
+
+    echo
+}
+
+install_prerequisites() {
+    echo "Checking if all prerequisites are installed..."
+
+    local -A packages=([java]=default-jre-headless [jq]=jq [nc]=netcat [netstat]=net-tools [sed]=sed [wget]=wget)
+
+    # Conditional packages
+    "$USE_HTTPS" && "$USE_LETSENCRYPT" && packages+=([certbot]=letsencrypt)
+    "$INSTALL_NGINX" && packages+=([nginx]=nginx-light)
+
+    # Check that packages exist
+    for key in "${!packages[@]}"; do
+        check_package "$key" "${packages[$key]}"
+    done
+
+    if [ "${#PACKAGES_TO_INSTALL[@]}" -ne 0 ]; then
+        echo "The following packages are missing and are now being installed: ${PACKAGES_TO_INSTALL[@]}"
+        echo "Installing them now..."
+        echo
+
+        sudo apt-get update
+        sudo apt-get install -y "${PACKAGES_TO_INSTALL[@]}"
+    else
+        echo "All packages installed!"
+    fi
+
+    check_nodejs
+
+    echo
+}
+
+calculate_variables() {
+    "$USE_HTTPS" || USE_LETSENCRYPT=false
+
+    PROTOCOL="$("$USE_HTTPS" && echo "https" || echo "http")"
+    EDITOR_URL="$PROTOCOL://$EXTERNAL_ADDRESS/"
+    BYTEBIN_URL="${EDITOR_URL}bytebin/"
+
+    if "$USE_HTTPS" && "$USE_LETSENCRYPT"; then
+        HTTPS_CERT_PATH="/etc/letsencrypt/live/$EXTERNAL_ADDRESS/fullchain.pem"
+        HTTPS_KEY_PATH="/etc/letsencrypt/live/$EXTERNAL_ADDRESS/privkey.pem"
+    fi
+
+    NGINX_LISTEN_DIRECTIVE_IPV4="$(get_nginx_sed_directive 4)"
+    NGINX_LISTEN_DIRECTIVE_IPV6="$(get_nginx_sed_directive 6)"
+}
+
+prepare_installation_location() {
+    echo "Now installing LuckPermsWeb..."
+    echo
+
+    if [ ! -d "$BASE_DIR" ]; then
+        sudo mkdir "$BASE_DIR"
+        sudo chown "$USER:$GROUP" "$BASE_DIR"
+    fi
+
+    cd "$BASE_DIR"
+}
+
+install_bytebin() {
+    echo "Installing bytebin..."
+
+    mkdir -p bytebin
+    pushd bytebin > /dev/null
+
+    # Find free port (stop the service if running)
+    sudo systemctl stop bytebin.service 2> /dev/null
+    BYTEBIN_PORT="$(find_free_port "$BYTEBIN_IP" "$BYTEBIN_PORT")"
+
+    # Download and Copy the Files
+    wget -q --show-progress --progress=dot:mega https://ci.lucko.me/job/bytebin/lastSuccessfulBuild/artifact/target/bytebin.jar
```

Comment:
--show-progress doesn't seem to work on my machine (centos 7) - could this be replaced with something else?